### PR TITLE
Refactor session transaction on sqlite store 

### PR DESCRIPTION
**Pull request by rashidakanchwala**
## Description

This ticket is a small refactor on the way we do session transactions on sqlite store. Previously we used different ways for session transactions i.e.
-  on RunsRepository we used context manager 
- on sqlite_store.py we use a session generator get_db() which was a bit odd. 

I have changed the sqlite_store.py to use context manager now just like how it is done in the Runs Repository. 

Besides this we have made a small change to database.py to do create the table metadata in create_db_engine function. 

Thanks @antonymilne for highlighting this and helping figure out :) 

## Checklist

- [ ] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro-viz/pull/1358"></a>

### CODE REVIEW

File: package/kedro_viz/integrations/kedro/sqlite_store.py [hunk 1]
```diff
@@ -35,6 +26,11 @@ def _is_json_serializable(obj: Any):
 class SQLiteStore(BaseSessionStore):
     """Stores the session data on the sqlite db."""
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        session_class = create_db_engine(self.location)
+        self._db_session = session_class
```
Comment (antonymilne) [inline package/kedro_viz/integrations/kedro/sqlite_store.py:32]:
```suggestion
        self._db_session = create_db_engine(self.location)
```
Super minor style improvement...

**Review by tynandebold — APPROVED** [review]
Excellent!

**Review by antonymilne — CHANGES_REQUESTED** [review]
I'm pretty sure this is ok, but I've been reading sqlalchemy docs some more and just want to check one thing... So please hold off on merging just for now!

**Review by antonymilne — APPROVED** [review]
@rashidakanchwala I just did a bit of re-naming and put in a new TODO.

This is a great change, but let's look at making further improvements straight after v1 of collaborative experiment tracking is done because I think there's a bit more reorganisation to do.

**Pull request merged**